plugins/woocommerce-blocks/tests/e2e/tests/reviews-by-product/reviews-by-product.block_theme.spec.ts

```diff
@@ -44,21 +40,51 @@ test.describe( `${ blockData.name } Block`, () => {
 		).toBeVisible();
 	} );
 
-	test( 'can change sort order in the frontend', async ( {
+	test( 'sorts by most recent by default and can sort by highest rating', async ( {
 		page,
 		frontendUtils,
+		editorUtils,
 	} ) => {
-		await page.goto( '/reviews-by-product/' );
+		await editorUtils.publishAndVisitPost();
+		const block = await frontendUtils.getBlockByName( BLOCK_NAME );
 
-		const block = await frontendUtils.getBlockByName( blockData.name );
-		let firstReview;
-		firstReview = block.locator( blockData.selectors.frontend.firstReview );
-		await expect( firstReview ).toHaveText( hoodieReviews[ 1 ].review );
+		const reviews = block.locator(
+			'.wc-block-components-review-list-item__text'
+		);
+
+		await expect( reviews.first() ).toHaveText( latestReview.review );
 
```

Comment:
It might be because we're mutating a global `allReviews` object by calling `sort` on it, e.g. here:

https://github.com/woocommerce/woocommerce/blob/d8800f331155278eadedcbcfb518ee922d7e9f8f/plugins/woocommerce-blocks/tests/e2e/tests/all-reviews/all-reviews.block_theme.spec.ts#L55-L57

Reply:
@WunderBart good spotting, totally forgot that sort is an inplace mutation. Will fix.

Reply:
Resolved https://github.com/woocommerce/woocommerce/pull/44880/commits/cd23e0dbc7743af6c0791af853749ec415ff0a77